pkg/controllers/gateway/gateway_controller.go

```diff
@@ -199,6 +201,57 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return reconcile.Result{}, r.Update(ctx, upstreamGateway)
 	}
 
+	var addressErr error
+	allAddresses := []gatewayv1beta1.GatewayAddress{}
+	for _, cluster := range clusters {
+		log.V(3).Info("checking cluster for addresses", "cluster", cluster)
+		addresses, addressErr := r.Placement.GetAddresses(ctx, upstreamGateway, cluster)
+		log.V(3).Info("got addresses", "addresses,", addresses, "addressErr", addressErr)
+		if addressErr != nil {
+			break
+		}
+		for _, address := range addresses {
+			log.V(3).Info("checking address type for mapping", "address.Type", address.Type)
+			var addressType gatewayv1beta1.AddressType
+			if *address.Type == gatewayv1beta1.IPAddressType {
+				addressType = MultiClusterIPAddressType
+			} else if *address.Type == gatewayv1beta1.HostnameAddressType {
+				addressType = MultiClusterHostnameAddressType
+			} else {
+				break // ignore address type. Unsupported for multi cluster gateway
+			}
+			allAddresses = append(allAddresses, gatewayv1beta1.GatewayAddress{
+				Type:  &addressType,
+				Value: fmt.Sprintf("%s/%s", cluster, address.Value),
+			})
+		}
+	}
+	if addressErr != nil {
+		return ctrl.Result{}, reconcileErr
+	}
+	log.V(3).Info("allAddresses", "allAddresses", allAddresses)
+	upstreamGateway.Status.Addresses = allAddresses
+
+	allListenerStatuses := []gatewayv1beta1.ListenerStatus{}
+	specListeners := upstreamGateway.Spec.Listeners
+	for _, listener := range specListeners {
+		for _, cluster := range clusters {
+			attachedRoutes, err := r.Placement.ListenerTotalAttachedRoutes(ctx, upstreamGateway, string(listener.Name), cluster)
+			if err != nil {
+				// May not have the status yet, let's ignore, but output info logs about it
+				log.Info("AttachedRoutes unknown for listener. Ignoring", "listener", listener.Name, "cluster", cluster, "message", err)
+				continue
+			}
+			allListenerStatuses = append(allListenerStatuses, gatewayv1beta1.ListenerStatus{
+				Name:           gatewayv1beta1.SectionName(fmt.Sprintf("%s.%s", cluster, string(listener.Name))),
```

Comment:
I think will need to be able to parse the cluster and listener name from this value, do we think just doing that based on a period delimiter will be sufficient? I assume you could create clusters or listeners with periods in the names which could cause issues.

Reply:
A [Listener SectionName](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.SectionName) can be a valid subdomain, so it could contain more than 1 period. The validation doesn't allow slashes or other symbols.

The [cluster name check](https://github.com/open-cluster-management-io/clusteradm/blob/main/pkg/cmd/join/preflight/checks.go#L97-L104) doesn't allow periods, slashes, underscores or other symbols except just hyphens.

Given we are bound by SectionName validation first, and the fact periods are not allowed in cluster names, I think having a period separator is reasonable.
To parse a multi cluster SectionName, you can split on the first period.
Can you think of anything we're missing here, or if there's a better way to handle this?

Reply:
> One caveat here is that cluster names with a hyphen in them will not pass the SectionName validation.
perhaps we need to allow for this edge case.

Despite this, that validation doesn't seem to be baked into the CRD validation, so it worked in my testing.
However it doesn't comply with the spec...

Reply:
Here's the validation pattern of a listener name in the crd

```
^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
```

> The validation doesn't allow slashes or other symbols

I was incorrect.
The validation allows hyphens

Reply:
To summarise the above, you can parse out the cluster name and listener name by splitting on the first period.
There may be more than 1 period, but that shouldn't be a problem if splitting stops at the first.

e.g. 

- mycluster.listenername => `mycluster` & `listenername`
- mycluster.listener.name => `mycluster` & `listener.name`
- my-cluster.listener.name => `my-cluster` & `listener.name`
- my-cluster.listener-name.org => `my-cluster` & `listener-name.org`
- my.cluster.listener.name => Invalid cluster name. Can't have periods. If this somehow happened, the cluster name would be assumed to be `my` with the listener name being `cluster.listener.name`


@mikenairn Given the above, are there any scenarios that are not allowed for?

Reply:
Yes, i think this is fine. I was just wondering when i seen it what issues we might hit. Seems though we should be fine as long as clusters stick to valid cluster names.